.github/workflows/e2e-nightly-ci.yml

```diff
@@ -28,7 +28,7 @@ env:
 
 jobs:
   e2e-fleet-nightly-test:
-    runs-on: ubuntu-latest
+    runs-on: runs-on,runner=8cpu-linux-x64,mem=16,run-id=${{ github.run_id }}
```

Comment:
Is this a leftover from tests? If not, why do we need to switch runners here?

Reply:
No, we need the internal runners because it [does not work](https://github.com/rancher/fleet/actions/workflows/e2e-nightly-ci.yml) on the default Github runners and I can not reproduce it. It works on the internal and my local machine though so I think we should be good.

I have also explained it in the commit.

```
test: Use internal runners for nightly builds

because on the Github ones test was failing all the time, while it
worked locally.
```